src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx

```diff
@@ -1,56 +1,440 @@
+      await queryClient.invalidateQueries({
+        queryKey: [reactQueries.thorchainLp.liquidityMember._def],
+        exact: false,
+        stale: true,
+      })
+      await queryClient.invalidateQueries({
+        queryKey: [reactQueries.thorchainLp.liquidityMembers._def],
+        exact: false,
+        stale: true,
+      })
+      await queryClient.invalidateQueries({
+        queryKey: [reactQueries.thorchainLp.userLpData._def],
+        exact: false,
+        stale: true,
+      })
+      await queryClient.invalidateQueries({
+        queryKey: [reactQueries.thorchainLp.liquidityProviderPosition._def],
+        exact: false,
+        stale: true,
+      })
+
+      setStatus(TxStatus.Confirmed)
+      onComplete()
+    },
+  })
+
+  useEffect(() => {
+    if (!(txId && tx)) return
+
+    if (tx?.status === TxStatus.Pending) {
+      setStatus(tx.status)
+      return
     }
 
-    return () => {
-      isMounted = false
+    // Avoids this hook's mutate fn running too many times
+    if (status === TxStatus.Confirmed) return
+
+    if (tx?.status === TxStatus.Confirmed) {
+      // The Tx is confirmed, but we still need to introspect completion from THOR itself
+      // so we set the status as pending in the meantime
+      setStatus(TxStatus.Pending)
+      ;(async () => {
+        await mutateAsync({ txId })
+      })()
+      return
     }
-  }, [onComplete, status])
+  }, [mutateAsync, onComplete, status, tx, txId])
+
+  const { data: inboundAddressData, isLoading: isInboundAddressLoading } = useQuery({
+    ...reactQueries.thornode.inboundAddress(assetId),
+    enabled: !!assetId,
+    select: data => data?.unwrap(),
+  })
+
+  const handleSignTx = useCallback(() => {
+    const isRuneTx = asset?.assetId === thorchainAssetId
+
+    if (
+      !(
+        assetId &&
+        poolAssetId &&
+        asset &&
+        poolAsset &&
+        wallet &&
+        (isRuneTx || inboundAddressData?.address)
+      )
+    )
+      return
+
+    return (async () => {
+      const accountId = isRuneTx ? runeAccountId : assetAccountId
+      if (!accountId) throw new Error(`No accountId found for asset ${asset.assetId}`)
+      const { account } = fromAccountId(accountId)
+      // TODO(gomes): rename the utils to use the same terminology as well instead of the current poolAssetId one.
+      // Left as-is for this PR to avoid a bigly diff
+      const thorchainNotationAssetId = assetIdToPoolAssetId({
+        assetId: isRuneTx ? poolAsset.assetId : asset.assetId,
+      })
+      const amountCryptoBaseUnit = toBaseUnit(amountCryptoPrecision, asset.precision)
+
+      // A THOR LP deposit can either be:
+      // - a RUNE MsgDeposit message type
+      // - an EVM custom Tx, i.e a Tx with calldata
+      // - a regular send with a memo (for ATOM and UTXOs)
+      const transactionType = (() => {
+        const supportedEvmChainIds = getSupportedEvmChainIds()
+        if (asset?.assetId === thorchainAssetId) return 'MsgDeposit'
```

Comment:
We might want to enum `MsgDeposit`, `EvmCustomTx`, and `Send`.

Reply:
Yeah I decided against using an enum here since this is a one-time use union and won't really be used anywhere else, TS will catch type errors for us as-is:

<img width="1037" alt="image" src="https://github.com/shapeshift/web/assets/17035424/22e43663-d2c1-46af-ba86-dc00771a2933">

Reply:
Perfect.